source/base/parameter_handler.cc

```diff
@@ -1617,12 +1617,19 @@ bool ParameterHandler::read_input (std::istream &input,
   std::string input_line;
   std::string fully_concatenated_line;
   bool is_concatenated = false;
+  // Maintain both the current line number and the current logical line
+  // number, where the later refers to the line number where (possibly) the
+  // current line continuation started.
   unsigned int current_line_n = 0;
-  bool status = true;
+  unsigned int current_logical_line_n = 0;
 
   while (std::getline (input, input_line))
     {
       ++current_line_n;
+      if (!is_concatenated)
+        {
+          current_logical_line_n = current_line_n;
+        }
```

Comment:
just remove the braces for this one-line statement. this gets more text onto one screen and makes it easier to read a whole algorithm. (at least that's my very personal opinion; you may disagree.)